packages/beeq/src/components/date-picker/bq-date-picker.tsx

```diff
@@ -0,0 +1,500 @@
+            </span>
+            {/* HTML Input */}
+            <input
+              id={this.name || this.fallbackInputId}
+              class="bq-date-picker__control--input"
+              autoComplete="off"
+              autoCapitalize="off"
+              autoFocus={this.autofocus}
+              aria-disabled={this.disabled ? 'true' : 'false'}
+              aria-controls={`${this.name}`}
+              aria-haspopup="dialog"
+              disabled={this.disabled}
+              form={this.form}
+              name={this.name}
+              placeholder={this.placeholder}
+              ref={(inputElem: HTMLInputElement) => (this.inputElem = inputElem)}
+              required={this.required}
+              spellcheck={false}
+              type="text"
+              value={this.value}
+              part="input"
+              // Events
+              onBlur={this.handleBlur}
+              onFocus={this.handleFocus}
+              onInput={this.handleInput}
+              onChange={this.handleChange}
+            />
+            {/* Clear Button */}
+            {this.hasValue && !this.disabled && !this.disableClear && (
+              // The clear button will be visible as long as the input has a value
+              // and the parent group is hovered or has focus-within
+              <bq-button
+                class="bq-date-picker__control--clear ms-[--bq-date-picker--gap] hidden"
+                appearance="text"
+                aria-label={this.clearButtonLabel}
+                size="small"
+                onBqClick={this.handleClearClick}
+                part="clear-btn"
+                exportparts="button"
+              >
+                <slot name="clear-icon">
+                  <bq-icon weight="bold" name="x-circle" class="flex" />
+                </slot>
+              </bq-button>
+            )}
+            {/* Suffix */}
+            <span
+              class="bq-date-picker__control--suffix"
+              ref={(spanElem: HTMLSpanElement) => (this.suffixElem = spanElem)}
+              part="suffix"
+            >
+              <slot name="suffix" onSlotchange={this.handleSuffixSlotChange}>
+                <bq-icon name="calendar-blank" weight="bold" class="flex" />
+              </slot>
+            </span>
+          </div>
+          <div class="flex items-center justify-center">
+            <CalendarComponentType
+              isDateDisallowed={this.isDateDisallowed}
+              locale={this.locale}
+              value={this.value}
+              min={this.min}
+              max={this.max}
+              months={this.months}
+              focusedDate={this.processFocusedDateValue(this.value)}
+              firstDayOfWeek={this.firstDayOfWeek}
+              showOutsideDays={this.showOutsideDays}
+              onChange={(ev: { target: { value: string } }) => {
+                this.value = ev.target.value;
+                this.open = !!this.multi;
+              }}
```

Comment:
I would recommend to move this into a private method

```ts
  private handleCalendarChange = (ev: { target: { value: string } }) => {
    this.value = ev.target.value;
    this.open = !!this.multi;
  };
  ...
  onChange={this.handleCalendarChange}
```

Reply:
Also, when the user selects a date from the calendar, the bqChange event is not emitted, meaning that we should also dispatch that event in the onChange of the Calendar.